docs/tla-plus/TransactionLayer/README.md

```diff
@@ -0,0 +1,86 @@
+3.***Key*** in KEYS(e.g., {"A","B"}).  
+4.The ***Value*** of each command is unique and can only be used when the operation is write.    
+5.For convenience, ***Transactions*** was defined at the begining, and use "exe_id" to indicate which command should be executed next.
+
+- A defined set of three transactions is a initial state, we can change an operation in a transaction to get a new initial state.  
+- E.g., if transaction-2 and transaction-3 is definite, and transaction-1 in specification is:  
+InitT1 ==/\ op1 \in {Write} /\ op2 \in {Read} /\ op3 \in {Write, Read}  
+　　　　/\ key1 \in {"A"}   /\ key2 \in {"A"} /\ key3 \in {"B"}  
+　　　　/\ T1 = << [op    |-> op1, key   |-> key1, value |-> 1],  
+　　　　　　　　　[op    |-> op2, key   |-> key2, value |-> 2],  
+　　　　　　　　　[op    |-> op3, key   |-> key3, value |-> 3]>>  
+Tansaction1 has 2 case :  
+　　Write A 1, Read A, Write B 3.  
+　　Write A 1, Read A, Read B.  
+Thus we have 2 initial state because of 2 cases of transaction-1.  
+
+## Use CRDB's strategy to handle these three transactions. 
+### 5 state transitions are defined in specification to handle transactions. It mainly includes:  
+- Begin a transaction and record information to transaction record(**BeginTransaction**).  
+- Using write pipeline to execute commands in every transations(**PipeLineWrite**).  
+- When all commands in a transaction is completed, its status should be changed to "staging"(**StartParallelCommit**), then check whether all inflight write have successfully persisted, if it is true, its status should be changed to "completed"(**CheckInflightAndCommit**).  
+- The write intent of the committed transaction can be persisted to MVCC data at any time(**CleanIntentWrite**).  
+
+### In the pipelinewrite section, transaction conflicts may occur, The processing logic is as follows:  
+#### Before read, you should know: for convenience, I made the following changes:  
+- When timestamp of a transaction is pushed, directly retry it, without refresh read or something else.
+- No crash scenarios were designed.
+- You can find more in :  https://www.cockroachlabs.com/docs/v21.2/architecture/transaction-layer#concurrency-control and https://github.com/cockroachdb/cockroach/blob/master/docs/design.md
+![framework](https://user-images.githubusercontent.com/62065552/146865735-ec42d028-d61f-45c0-acb9-86a24472180d.png)
+  
+- Handle conflict 1 and handle conflict 2 will be described in detail below.
+
+#### Handle conflict 1
+![readconflict](https://user-images.githubusercontent.com/62065552/146737545-206f312f-47c9-4fc8-b17c-0562c1d83cd6.png)
+
+#### Handle conflict 2
+![writeconflict](https://user-images.githubusercontent.com/62065552/146737690-ca539b29-876c-4500-854c-6c608c5c0656.png)
+
+## Check invariants and properties during and at the end of the run.  
+There are many invariants and properties, and the most important one is:  
+***Strategy of handling conflict is correct.***
+  
```

Comment:
Because of SSI , the result should be one of the six sort order(123,132,231,213,321,312) execution results of three transactions. 
Further more，we can use timestamp to choose which one is expected in these  six sort order execution results.